runtime-sdk/src/modules/consensus_accounts/test.rs

```diff
@@ -0,0 +1,412 @@
+    let mut mock = mock::Mock::default();
+    let mut ctx = mock.create_ctx();
+    let mut meta = Metadata {
+        ..Default::default()
+    };
+    let genesis = Default::default();
+
+    Module::<Accounts, Consensus>::init_or_migrate(&mut ctx, &mut meta, &genesis);
+
+    let tx = transaction::Transaction {
+        version: 1,
+        call: transaction::Call {
+            method: "consensus.Deposit".to_owned(),
+            body: cbor::to_value(Deposit {
+                amount: BaseUnits::new(1_000.into(), Denomination::NATIVE),
+            }),
+        },
+        auth_info: transaction::AuthInfo {
+            signer_info: vec![transaction::SignerInfo::new(keys::alice::pk(), 0)],
+            fee: transaction::Fee {
+                amount: Default::default(),
+                gas: 1000,
+            },
+        },
+    };
+
+    ctx.with_tx(tx, |mut tx_ctx, call| {
+        assert!(Module::<Accounts, Consensus>::tx_deposit(
+            &mut tx_ctx,
+            cbor::from_value(call.body).unwrap(),
+        )
+        .is_err());
+    });
+}
+
+#[test]
+fn test_api_deposit() {
+    let mut mock = mock::Mock::default();
+    let mut ctx = mock.create_ctx();
+    let mut meta = Metadata {
+        ..Default::default()
+    };
+    let genesis = Default::default();
+
+    Module::<Accounts, Consensus>::init_or_migrate(&mut ctx, &mut meta, &genesis);
+
+    let tx = transaction::Transaction {
+        version: 1,
+        call: transaction::Call {
+            method: "consensus.Deposit".to_owned(),
+            body: cbor::to_value(Deposit {
+                amount: BaseUnits::new(1_000.into(), Denomination::from_str("TEST").unwrap()),
+            }),
+        },
+        auth_info: transaction::AuthInfo {
+            signer_info: vec![transaction::SignerInfo::new(keys::alice::pk(), 0)],
+            fee: transaction::Fee {
+                amount: Default::default(),
+                gas: 1000,
+            },
+        },
+    };
+
+    ctx.with_tx(tx, |mut tx_ctx, call| {
+        Module::<Accounts, Consensus>::tx_deposit(
+            &mut tx_ctx,
+            cbor::from_value(call.body).unwrap(),
+        )
+        .expect("deposit tx should succeed");
+
+        let (_, msgs) = tx_ctx.commit();
+        assert_eq!(1, msgs.len(), "one message should be emitted");
+        let (msg, hook) = msgs.first().unwrap();
+
+        assert_eq!(
+            &Message::Staking {
+                v: 0,
+                msg: StakingMessage::Withdraw(staking::Withdraw {
+                    from: keys::alice::address().into(),
+                    amount: 1_000.into(),
+                })
+            },
+            msg,
+            "emitted message should match"
+        );
+
+        assert_eq!(
+            CONSENSUS_WITHDRAW_HANDLER.to_string(),
+            hook.hook_name,
+            "emitted hook should match"
+        )
+
+        // TODO: support advancing the round. And ensure message is correctly processed.
```

Comment:
Would be useful to have support for this in the testing framework. Shouldn't be hard to add, maybe open an issue?

Reply:
Done https://github.com/oasisprotocol/oasis-sdk/issues/110